File: Assets/Scripts/Core/Deck.cs
using System.Collections.Generic;
using UnityEngine;

namespace Core
{
    public class Deck : MonoBehaviour
    {
        [Header("Configuration")]
        public GameObject cardPrefab;
        public List<Sprite> cardSprites; // Assume ordered: 0-12 Clubs, 13-25 Diamonds, etc.

        private List<Card> cards = new List<Card>();
        private int currentIndex = 0;

        public void InitializeDeck()
        {
            // Clear existing if any (logic for reset)
            foreach(Transform child in transform)
            {
                Destroy(child.gameObject);
            }
            cards.Clear();

            // Create 52 cards
            int spriteIndex = 0;
            foreach (Suit suit in System.Enum.GetValues(typeof(Suit)))
            {
                foreach (Rank rank in System.Enum.GetValues(typeof(Rank)))
                {
                    GameObject cardObj = Instantiate(cardPrefab, transform);
                    Card cardScript = cardObj.GetComponent<Card>();
                    
                    Sprite faceSprite = (spriteIndex < cardSprites.Count) ? cardSprites[spriteIndex] : null;
                    
                    cardScript.Setup(suit, rank, faceSprite);
                    cardObj.SetActive(false); // Hide in deck
                    
                    cards.Add(cardScript);
                    spriteIndex++;
                }
            }
            
            Shuffle();
        }

        public void Shuffle()
        {
            // Fisher-Yates Shuffle
            int n = cards.Count;
            while (n > 1)
            {
                n--;
                int k = Random.Range(0, n + 1);
                Card value = cards[k];
                cards[k] = cards[n];
                cards[n] = value;
            }
            currentIndex = 0;
        }

        public Card DrawCard()
        {
            if (currentIndex >= cards.Count)
            {
                Debug.LogWarning("Deck empty! Reshuffling.");
                Shuffle();
            }

            Card card = cards[currentIndex];
            currentIndex++;
            return card;
        }
        
        public void ResetDeck()
        {
            foreach(var c in cards)
            {
                c.gameObject.SetActive(false);
                c.transform.SetParent(transform);
            }
            Shuffle();
        }
    }
}

File: js/core/Deck.js
// Card is globally available via window.Card

window.Deck = class Deck {
    constructor() {
        this.cards = [];
        this.suits = ['Clubs', 'Diamonds', 'Hearts', 'Spades'];
        this.ranks = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A'];
        this.generateDeck();
    }

    generateDeck() {
        this.cards = [];
        for (let suit of this.suits) {
            for (let rank of this.ranks) {
                let value = parseInt(rank);
                if (rank === 'J' || rank === 'Q' || rank === 'K') value = 10;
                if (rank === 'A') value = 11; // Hand logic manages Ace value

                this.cards.push(new Card(suit, rank, value));
            }
        }
    }

    shuffle() {
        // Fisher-Yates
        for (let i = this.cards.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [this.cards[i], this.cards[j]] = [this.cards[j], this.cards[i]];
        }
    }

    draw() {
        if (this.cards.length === 0) {
            console.log("Deck empty, reshuffling new deck.");
            this.generateDeck();
            this.shuffle();
        }
        return this.cards.pop();
    }
}
